src/Text/Gigaparsec/Internal/Token/Numeric.hs

```diff
@@ -371,23 +371,35 @@ mkSigned NumericDesc{..} !unsigned !err = IntegerParsers {
         _number = annotate (labelIntegerSignedNumber err Nothing) $
           atomic (sign <*> annotate (labelIntegerNumberEnd err) (number unsigned))
 
-{-type FloatingParsers :: *
-data FloatingParsers = FloatingParsers {}
+type FloatingParsers :: *
+data FloatingParsers = FloatingParsers { 
+  -- | Parse a single real number literal in decimal form (base 10).
+    decimalFloating :: Parsec Rational
+  -- | Parse a single real number literal in hexadecimal form (base 16).
+  , hexadecimalFloating :: Parsec Rational
+  -- | Parse a single real number literal in octal form (base 8).
+  , octalFloating :: Parsec Rational
+  -- | Parse a single real number literal in binary form (base 2).
+  , binaryFloating :: Parsec Rational
+  -- | Parse a single real number literal, 
+  -- which can be in many forms and bases depending on the configuration.
+  , numberFloating :: Parsec Rational
+  }
 
-mkUnsignedFloating :: NumericDesc -> IntegerParsers CanHoldUnsigned -> GenericNumeric -> FloatingParsers
-mkUnsignedFloating NumericDesc{..} nat gen = FloatingParsers {}
+-- mkUnsignedFloating :: NumericDesc -> IntegerParsers CanHoldUnsigned -> GenericNumeric -> ErrorConfig -> FloatingParsers
+-- mkUnsignedFloating NumericDesc{..} nat gen err = FloatingParsers {}
```

Comment:
To implement this, should I create a helper function similar to `ofRadix` in "UnsignedReal.scala", then use it to define the various fields in `FloatingParsers`?

Reply:
Probably? What does the Scala code do?

Reply:
According to my understanding of [the code](https://github.com/j-mie6/parsley/blob/staging%2F5.0/parsley%2Fshared%2Fsrc%2Fmain%2Fscala%2Fparsley%2Ftoken%2Fnumeric%2FUnsignedReal.scala), there's a `private def ofRadix(radix: Int, digit: Parsley[Char], endLabel: LabelConfig)` which does all the heavy work, and `_decimal`, `_hexadecimal` etc. all uses it.

Reply:
Then go with that then :)